### PR TITLE
Adding simple customer promt screen

### PR DESCRIPTION
**Pull request by girish-lokapure**
dependency:  https://github.com/newfold-labs/wp-module-onboarding-data/pull/20

### CODE REVIEW

File: src/OnboardingSPA/steps/SiteGen/SiteDetails/walkthrough/index.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+	const selectOption = ( event, classname ) => {
+		document.querySelectorAll( `.${ classname }` ).forEach( ( button ) => {
+			button.classList.remove(
+				'nfd-sg-site-details-rows-button-selected'
+			);
+		} );
+		event.target.classList.add(
+			'nfd-sg-site-details-rows-button-selected'
+		);
```
Comment (officiallygod) [inline src/OnboardingSPA/steps/SiteGen/SiteDetails/walkthrough/index.js:126]: Let's change the classes with React code using `classnames` and not Directly manipulate the DOM.

**Review by officiallygod — CHANGES_REQUESTED** [review]
https://github.com/newfold-labs/wp-module-onboarding/assets/48691514/c0e247c4-40fb-405b-8c1c-515d7a7012d4

- There is a BUG that needs to be addressed and THE DATA IS NOT SAVED.

Screenshot 2023-12-13 at 3 49 21 PM

- Please add Bottom Margin in Mobile View.

Screenshot 2023-12-13 at 3 56 14 PM

- NEXT Needs to be disabled as it was done on the Site Details page.

Screenshot 2023-12-13 at 3 48 21 PM

- I think the Text should be WHITE and not gray after the user types it.

Screenshot 2023-12-13 at 3 51 39 PM

- Let's remove unnecessary spaces and fix typos to make it better formatted for the user.

Thanks.

File: src/OnboardingSPA/components/Button/ButtonWhite/SiteGen/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Button } from '@wordpress/components';
+import classNames from 'classnames';
+
+const ButtonSiteGen = ( { className, value, text, onClick } ) => {
+	return (
+		<Button
+			className={ classNames( className ) }
```
[the remark this answers is not in the document]
Comment (girish-lokapure) [reply src/OnboardingSPA/components/Button/ButtonWhite/SiteGen/index.js:7]: we will be having multiple classnames

File: src/OnboardingSPA/steps/SiteGen/SiteDetails/walkthrough/index.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+import { getSiteDetailsmeta } from '../../../../utils/api/siteGen';
+
+const SiteGenSiteDetailsWalkthrough = () => {
+	const content = getContents();
+	const isLargeViewport = useViewportMatch( 'small' );
+	const [ isEditing, setEditing ] = useState( false );
+	const [ customerInputName, setCustomerInputName ] = useState();
+	const [ customerInputType, setCustomerInputType ] = useState();
+	const [ customerInputStyle, setCustomerInputStyle ] = useState();
+	const [ customerInputUnique, setCustomerInputUnique ] = useState();
```
[the remark this answers is not in the document]
Comment (girish-lokapure) [reply src/OnboardingSPA/steps/SiteGen/SiteDetails/walkthrough/index.js:20]: I think this is fine, the load/lag was due to the api call which has been fixed

**Review by arunshenoy99 — CHANGES_REQUESTED** [review]
image
Bug: The placeholder text appears in white, giving the impression that the input is filled. Can we match it to Figma's specifications?

https://github.com/newfold-labs/wp-module-onboarding/assets/38878906/83036744-0055-4370-a031-99a975f84113
Bug: The selected button grays out, giving a false indication that another option was selected. Can we align this with Figma's specifications?

image
Bug: Can we add some padding between the text and the textarea border?

image
Bug: I cannot proceed to the next screen when in this state. Additionally, the highlighted white color suggests 'Yes' was selected, but I have actually chosen 'No,' which is now grayed out.

image
Could we add an appropriate placeholder for this input?

https://github.com/newfold-labs/wp-module-onboarding/assets/38878906/01732971-2644-4187-8545-055ab6f27e4c
When we go back, should we return to the simple prompt screen or the welcome screen?

image
Bug: In the mobile view, the textarea blurs into the footer. Can we add some spacing or adjust the layout to address this issue?

I will review more once these issues are addressed. Thanks, @girish-lokapure.

File: includes/RestApi/SiteGenController.php [hunk 4]
```diff
@@ -106,4 +115,42 @@ public function generate_sitegen_meta( \WP_REST_Request $request ) {
...
+				],
+				[
+					"question" => "4. Is there anything unique about your business or brand?",
+					"field" => "uniqueBusiness",
+					"placeholder" => "e.g. Unique product, amazing customer service, customizations, etc.",
+					"prompt" => "Unique about my business is ",
+				],
+			]
+		);
+	}
```
[the remark this answers is not in the document]
Comment (girish-lokapure) [reply includes/RestApi/SiteGenController.php]: moved to onboarding data

File: src/OnboardingSPA/steps/SiteGen/SiteDetails/contents.js [hunk 5]
```diff
@@ -12,6 +12,19 @@ const getContents = () => {
...
+		),
+		walkThroughlink: __( 'click here', 'wp-module-onboarding' ),
+		writeStyleOption1: __(
+			'We craft awesome goodies!',
+			'wp-module-onboarding'
+		),
+		writeStyleOption2: __(
+			'We manufacture quality products',
+			'wp-module-onboarding'
+		),
```
[the remark this answers is not in the document]
Comment (girish-lokapure) [reply src/OnboardingSPA/steps/SiteGen/SiteDetails/contents.js:27]: As discussed this is not required now

**Comment by girish-lokapure** [comment]
- fixed Flow data API 
- Input sanitized and trimmed
- Modified Color schema for light mode
- Fixed navigation “Next” button issues

**Review by arunshenoy99 — APPROVED** [review]

**Review by arunshenoy99 — APPROVED** [review]

**Pull request merged**